pkg/utils/pools/pools_test.go

```diff
@@ -53,3 +53,108 @@ func TestPool(t *testing.T) {
 		t.Errorf("expected 3, got %d", pool.Get())
 	}
 }
+
+func TestPool_Reuse(t *testing.T) {
+	index := 0
+	pool := NewPool(func() int {
+		index++
+		return index
+	})
+
+	// Get multiple values
+	a := pool.Get()
+	b := pool.Get()
+	c := pool.Get()
+
+	// Put them back in a different order
+	pool.Put(b)
+	pool.Put(a)
+	pool.Put(c)
+
+	// Create a map to keep track of seen values
+	seen := make(map[int]bool)
+
+	// Check the reuse of the pool
+	seen[pool.Get()] = true
+	seen[pool.Get()] = true
+	seen[pool.Get()] = true
+
+	// Ensure we have seen all values
+	if !seen[1] || !seen[2] || !seen[3] {
+		t.Errorf("expected to see 1, 2, and 3, but got seen map: %v", seen)
+	}
+}
+
+func TestPool_ConcurrentAccess(t *testing.T) {
+	pool := NewPool(func() int {
+		return 0
+	})
```

Comment:
```suggestion
	var index int32
	pool := NewPool(func() int {
		return int(atomic.AddInt32(&index, 1))
	})
```

and check that the results are not duplicated